SaltSquareIO/sports-app-react-fe
# Unauthorized access

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0g7n7z)

## Summary of change

This PR implements an authentication mechanism and protected routing to meet the requirements for secure and consistent user access to the application. Key changes include: 

1. Ensures that authenticated users can access the home page (`BASE_URL`) and profile page (`BASE_URL/profile`) directly when logged in.
2. Implements automatic redirection for non-authenticated users accessing `BASE_URL` and `BASE_URL/profile` to the login page.
3. Includes persistent session handling so that users remain logged in when they close and reopen the app.
4. Configures the ProtectedRoute component to handle route protection for `BASE_URL` and `BASE_URL/profile`.

## QA Notes

To test this PR, follow these instructions:

1. Access `BASE_URL` and `BASE_URL/profile` without logging in to verify redirection to the login page.
2. Log in to the application and navigate to `BASE_URL` and `BASE_URL/profile` to verify that access is permitted.
3. After logging in, close and reopen the browser, then navigate to `BASE_URL` and `BASE_URL/profile` to confirm that the session persists. 
4. To simulate token expiration, clear cookies. Attempt navigation to protected routes to confirm redirection to login.
